### ISSUE TITLE
Follow up for PR https://github.com/project-chip/connectedhomeip/pull/29364

### ISSUE THREAD

**Issue by nivi-apple** [project-chip/connectedhomeip#36526]
see - https://github.com/project-chip/connectedhomeip/pull/29364/files#r1826271763

**Comment by nivi-apple** [comment]
also address add tests with multiple controllers

**Comment by nivi-apple** [comment]
Fix 
So... this part is wrong. If mOTAImageTransferHandler is already not null, we will clobber the pointer and then things will go very badly wrong on shutdown. The fact that we apparently have no tests exercising this case is very bad and we need to fix that.

We need to do one of two things:

Actually support multiple instances of MTROTAImageTransferHandler being live, and then we can create a new one here, or:
Have a check for IsInAnOngoingTransfer() and fail out if it's true here.

Note that this is for the "we got an unsolicited BDX message" case. The reason our tests work is that we do respond BUSY from the QueryImage command if we are in a transfer, which would be before the other side tries to start BDX, so in "normal" operation we never reach this code with IsInAnOngoingTransfer() true.

So long story short, for now

Put a IsInAnOngoingTransfer() check at the top of this method and error out if it's true.
Do not set mOTAImageTransferHandler here. It gets set in OnTransferHandlerCreated.
As a followup, we start supporting multiple MTROTAImageTransferHandler being live.

- https://github.com/project-chip/connectedhomeip/pull/29364/files#r1849059218
